breacharbiter.go

```diff
@@ -347,46 +390,39 @@ out:
 			b.wg.Add(1)
 			go b.exactRetribution(cfChan, breachInfo)
 
-			delete(b.breachObservers, breachInfo.chanPoint)
+			delete(breachObservers, breachInfo.chanPoint)
 
 		case chanPoint := <-b.newContracts:
 			// A new channel has just been opened within the
 			// daemon, so we launch a new breachObserver to handle
 			// the detection of attempted contract breaches.
-			settleSignal := make(chan struct{})
+			breachSignal := make(chan *ContractBreachEvent, 1)
 
 			// If the contract is already being watched, then an
 			// additional send indicates we have a stale version of
 			// the contract. So we'll cancel active watcher
 			// goroutine to create a new instance with the latest
 			// contract reference.
-			if oldSignal, ok := b.breachObservers[chanPoint]; ok {
+			if oldSignal, ok := breachObservers[chanPoint]; ok {
 				brarLog.Infof("ChannelPoint(%v) is now live, "+
 					"abandoning state contract for live "+
 					"version", chanPoint)
 				close(oldSignal)
 			}
 
-			b.breachObservers[chanPoint] = settleSignal
+			breachObservers[chanPoint] = breachSignal
 
 			brarLog.Debugf("New contract detected, launching " +
 				"breachObserver")
 
-			chainEvents, err := b.cfg.SubscribeChannelEvents(chanPoint)
-			if err != nil {
-				// TODO(roasbeef); panic?
-				brarLog.Errorf("unable to register for event "+
-					"sub for chan_point=%v: %v", chanPoint, err)
-			}
-
 			b.wg.Add(1)
-			go b.breachObserver(chainEvents, settleSignal)
+			go b.breachObserver(chanPoint, breachSignal)
 
 		case chanPoint := <-b.settledContracts:
```

Comment:
This case should now be unnecessary since all unilateral/coop closes are handled directly from the subscription channel, and we wouldn't signal this from the handoff task